ext/intl/grapheme/grapheme_string.c

```diff
@@ -179,9 +177,9 @@ PHP_FUNCTION(grapheme_stripos)
 		char *haystack_dup, *needle_dup;
 		int32_t noffset = offset >= 0 ? offset : (int32_t)haystack_len + offset;
 		needle_dup = estrndup(needle, needle_len);
-		php_strtolower(needle_dup, needle_len);
+		zend_str_tolower(needle_dup, needle_len);
```

Comment:
What's the reasoning for changing grapheme? (I've never used grapheme)

I'm not familiar with what this extension is doing for the utf-16 case? But it seems like the utf-16 case is locale dependent (on the system locale?) (grapheme_get_break_iterator, grapheme_strpos_utf16), and if utf-16 is locale-dependent, ascii should be (as close to possible) as well, to work consistently in turkish for utf-8 inputs with non-ascii codepoints.

For foreign locales (e.g. accents, turkish, etc), keeping the locale dependence seems useful 

https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/classicu_1_1Locale.html#a020c6966493a8f00572616b64b5527c3

global_break_iterator is a singleton initialized once with system default locale? Not 100% sure if that's what "icu default locale" means

```
		global_break_iterator = ubrk_open(UBRK_CHARACTER,
											NULL,	/* icu default locale - locale has no effect on this iterator */
											NULL,	/* text not set in global iterator */
											0,		/* text length = 0 */
											status);

		INTL_G(grapheme_iterator) = global_break_iterator;
```

Reply:
Most of the constructors in ext/intl have a locale parameter. The grapheme functions are unusual in not having an explicit locale dependency. So my expectation as a user reading the documentation is that these functions should do Unicode Default Case Conversion. As such, I think my change here is correct.

The other possible interpretation of the manual is that intl.default_locale should be used. I can make it do that if that is desired. I don't think it should depend on the C library locale or the environment variables.

I tested the old and new behaviour. usearch_open() is called with locale="", but that is not interpreted as the default locale, it's interpreted as a locale in which all the fields are empty strings and the behavior is fixed.

Using tolower() is not correct in any case, since it can't do UTF-8, and the input string is always UTF-8. The effect of using tolower() with a Turkish locale is to mangle ASCII input, preventing `i` from matching `I`, which you might almost call a benefit, but the lack of locale sensitivity in the slow path means that `i` won't match `İ` either.

The break iterator is not doing case folding so the locale attached to it is not important.

Reply:
> I tested the old and new behaviour. usearch_open() is called with locale="", but that is not interpreted as the default locale, it's interpreted as a locale in which all the fields are empty strings and the behavior is fixed.

> The break iterator is not doing case folding so the locale attached to it is not important.

Thanks, I was just mistaken and had problems understanding what the code was currently doing, not wanting a change to the functionality.